### PR TITLE
applications: Matter Bridge: make adding new devices more flexible

### PR DESCRIPTION
**Pull request by markaj-nordic**
This patch allows to add different Matter devices tied to the same data provider
in multiple steps - up till now it has to be added in a single call.

### CODE REVIEW

**Review by LuDuda — APPROVED** [review]

File: samples/matter/common/src/bridge/bridge_manager.cpp [hunk 1]
```diff
@@ -176,6 +178,18 @@ CHIP_ERROR BridgeManager::AddSingleDevice(MatterBridgedDevice *device, BridgedDe
 {
 	uint8_t index{ 0 };
 	CHIP_ERROR err;
+	uint16_t duplicatedItemKeys[kMaxBridgedDevices];
+	BridgedDevicePair pair(device, dataProvider);
+
+	/* Check if the current provider is already contained in the map - if so, there is at least one duplicate */
+	bool isNewProvider = (Instance().mDevicesMap.GetDuplicates(pair, duplicatedItemKeys) == 0);
```
Comment (kkasperczyk-no) [inline samples/matter/common/src/bridge/bridge_manager.cpp:185]: Nit: do we need to allocate a variable for it? Maybe we could just move this check to the if statement two lines below.

File: samples/matter/common/src/bridge/util/bridge_util.h [hunk 2]
```diff
@@ -125,21 +125,20 @@ template <typename T, std::size_t N> struct FiniteMap {
 	}
 	std::size_t FreeSlots() { return N - mElementsCount; }
 
-	bool ValueDuplicated(const T &value, uint16_t *key, uint8_t &numberOfDuplicates)
+	uint8_t GetDuplicates(const T &value, uint16_t *key)
```
Comment (kkasperczyk-no) [inline samples/matter/common/src/bridge/util/bridge_util.h:128]: How about `GetDuplicatesCount or Number`? Currently the name suggest we would get e.g. list of duplicated objects.

**Review by kkasperczyk-no — APPROVED** [review]
Left two minor comments, but LGTM overall.

File: samples/matter/common/src/bridge/bridge_manager.cpp (hunk 1 above)
Comment (markaj-nordic) [reply samples/matter/common/src/bridge/bridge_manager.cpp:185]: I wanted to achieve more self-documented code :) I'd like to keep it that way if you don't mind. Depending on the returned number of duplicates we may have different logic in the application, so I guess explicitness, even if redundant, does not hurt.

**Pull request merged**